### PR TITLE
file utils: introduce get_host_tool_path() to find commands on host

### PR DESCRIPTION
**Pull request by cjp256**
A counterpart to get_snap_tool_path() for finding commands inside of
snapcraft snap, this finds tools required on the host.

Raises snapcraft.internal.errors.SnapcraftHostToolNotFoundError if not found.

- Renames HostToolNotFoundError -> SnapcraftHostToolNotFoundError

- Uses this new method for lifecycle finding snapd.

- Update tests (and minor refactor to import 'errors' rather
  than continue cherry-picking each exception import.)

Signed-off-by: Chris Patterson <chris.patterson@canonical.com>

- [ ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

File: snapcraft/cli/lifecycle.py [hunk 1]
```diff
@@ -201,11 +201,10 @@ def _run_pack(snap_command: List[str]) -> str:
...
     directory: str, *, compression: Optional[str] = None, output: Optional[str]
 ) -> None:
-    snap_path = shutil.which("snap")
-    if snap_path is None:
-        raise errors.HostToolNotFoundError(command_name="snap", package_name="snapd")
 
-    command = [snap_path, "pack"]
+    snap_path = file_utils.get_host_tool_path(command_name="snap", package_name="snapd")
+
+    command = [str(snap_path), "pack"]
```
Comment (sergiusens) [inline snapcraft/cli/lifecycle.py:207]: I don't think str is not entirely necessary here, subprocess does the right thing

File: snapcraft/file_utils.py [hunk 2]
```diff
@@ -335,6 +330,26 @@ def calculate_hash(path: str, *, algorithm: str) -> str:
...
 
 
+def get_host_tool_path(*, command_name: str, package_name: str) -> pathlib.Path:
+    """Return the path command found on the host.
+
+    Using shutil.which() to find the host-provided command,
+    raises SnapcraftHostMissingToolError if not found.
+
+    :param command_name: command to find
+    :param package_name: package <command> is usually found in
```
Comment (sergiusens) [inline snapcraft/file_utils.py:340]: shouldn't `<command>` be just `command_name` in this docstring entry?

File: snapcraft/file_utils.py [hunk 3]
```diff
@@ -335,6 +330,26 @@ def calculate_hash(path: str, *, algorithm: str) -> str:
...
+    :raises SnapcraftHosterrors.ToolMissingError: if command not found.
```
Comment (sergiusens) [inline snapcraft/file_utils.py:341]: might want to revisit this exception docstring

File: tests/unit/test_file_utils.py [hunk 4]
```diff
@@ -330,10 +326,24 @@ def test_get_tool_from_docker_snap_path(
...
 
+def test_get_host_tool_finds_command(monkeypatch):
+    monkeypatch.setattr(shutil, "which", lambda x: "/usr/bin/foo")
+
+    assert file_utils.get_host_tool_path(command_name="foo", package_name="foo")
+
+
+def test_get_host_tool_failure(monkeypatch):
+    monkeypatch.setattr(shutil, "which", lambda x: None)
+    with pytest.raises(errors.SnapcraftHostToolNotFoundError) as error:
```
Comment (sergiusens) [inline tests/unit/test_file_utils.py:337]: this is just a nit about an empty line between test setup and actual test

File: snapcraft/cli/lifecycle.py (hunk 1 above)
Comment (cjp256) [reply snapcraft/cli/lifecycle.py:207]: Hrm, you're right indeed.  I don't see that in the docs anywhere but some googling shows that it should work, and where it doesn't work in subprocess has been treated like a bug.

File: tests/unit/test_file_utils.py (hunk 4 above)
Comment (cjp256) [reply tests/unit/test_file_utils.py:338]: done

File: snapcraft/file_utils.py (hunk 3 above)
Comment (cjp256) [reply snapcraft/file_utils.py]: done

File: snapcraft/file_utils.py (hunk 2 above)
Comment (cjp256) [reply snapcraft/file_utils.py]: done

File: snapcraft/cli/lifecycle.py (hunk 1 above)
Comment (cjp256) [reply snapcraft/cli/lifecycle.py]: done

**Pull request merged**